### PR TITLE
Allow empty argument in functional pseudos

### LINKED ISSUE #219: Empty argument pseudos such as `:has()` throw
When parsing `:has` with the context of a `selector` it works fine:

```json
{
    "type": "Selector",
    "loc": null,
    "children": [
        {
            "type": "PseudoClassSelector",
            "loc": null,
            "name": "has",
            "children": null
        }
    ]
}
```

When parsing `:has()` however, it does not:

```
❌ SyntaxError: Selector is expected
```

I expected the output here to be the same in both cases. Is this something that’s feasible to be fixed? I can take a look into it if wanted.

Demo: https://codepen.io/bramus/pen/NWzQqdp/c8766b9b857cf1637f254a16269a5848

### PR DESCRIPTION
**Pull request by bramus**
Fixes #219.

Marking this PR as draft for now, as I’m not sure what to do `nth-child` and its variants. Should relaxed parsing – allowing an empty argument – also be allowed there? If so, it’s a matter of changing the fixture to not mark `""` as invalid.

### CODE REVIEW

**Comment by lahmatiy** [comment]
@bramus 
> If so, it’s a matter of changing the fixture to not mark "" as invalid.

A failed test is a little weird. It tests parsing `<an-plus-b>` but through `:nth-child()` pseudo. So empty string is invalid for `<an-plus-b>`, but that's fine for `:nth-child()` content. I think you may change the test to ignore empty string case in `fixture.invalid.forEach` block.

Everything else looks good to me 👍

**Comment by bramus** [comment]
Updated tests to reflect resolution as per https://github.com/csstree/csstree/pull/220#issuecomment-1353438706

File: fixtures/definition-syntax-match/generic.json [hunk 1]
```diff
@@ -457,7 +457,6 @@
             "123456n-12345678"
         ],
         "invalid": [
-            "",
```
Comment (lahmatiy) [inline fixtures/definition-syntax-match/generic.json:460]: This line shouldn't be removed. This fixture is for testing definition syntax and an empty string is invalid. I commented before that a test which testing `An+B` parsing should be altered to exclude this case. The test I referred is actually re-using this fixture for its own purposes.

**Comment by lahmatiy** [comment]
@bramus Thank you!

**Pull request merged**